core/contractHost.chainmail

```diff
@@ -0,0 +1,113 @@
+/**
+ * ContractHost provides a way to install and run verifiable contracts.
+ *
+ * Install(source) creates Installations. Installations represent particular
+ * contract forms, which can be spawn()'d to create Contract instances
+ * initialized with specific terms. These Contracts manage invitations for
+ * multiple seats, each of which repesents a role in an transaction. After a
+ * role has carried out its responsibilities, it may collect (q.v.) winnings and
+ * refunds into appropriate purses.
+ */
+interface ContractHost {
+  /**
+   * ContractSrc is the source code for a Contract. See Contract for details.
+   */
+  install(contractSrc:Text) -> (Installation);
+
+  /** Returns the actual source of the Installation. */
+  getInstallationSourceCode(installation:Installation) -> (Text);
+
+  /**
+   * Seat invitations are packaged as payments from the Invite
+   * Issuer. Redeeming an invite returns a role object with an arbitrary
+   * interface (the interface is at the discretion of the contract) which
+   * supports interaction according to the terms.
+   */
+  redeem(invite:Payment) -> (Object); // can we say more about the returned value?
```

Comment:
> Use right is like membership card: while you have it you can use without consuming it. 

I think we can make the use right work however we would like - i.e. it can be consumed. All we would have to do is add that aspect to the makeUseObj function passed into makeMint.